x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,222 @@
+---
+template:
+  settings:
+    # This is a made up setting which does not exist today. The idea is that
+    # any field that shows up after the first 1000 fields is kept in source
+    # but does not show up as runtime for index field
+    #index.mapping.total_fields.soft_limit: 1000
+
+    # Ignores any malformed data. Data will still be persisted in _source
+    index.mapping.ignore_malformed": true
+
+  mappings:
+
+    # Any field that is discovered is mapped as a runtime field
+    dynamic: runtime
+
+    # DISCUSS: What are the pros / cons of having this enabled?
+    date_detection: false
+
+    ### All the properties which are set in each template
+    properties:
+
+      # This ensures all subobjects are flattened and prevents and object / keyword conflicts
+      # As subobject is set to false, no objects like host, error or similar are mapped.
+      subobject: false
+
+      # @timestamp is the only field that MUST exist.
+      "@timestamp":
+        type: date
+        ignore_malformed: false
+
+      # The data stream properties are required in any index following the data stream naming scheme
+      # Discuss: If there is an ecs: enabled flag it can also be used in indices so this would have to be skipped
+      # These fields might not even be needed in this template
+      data_stream.type:
```

Comment:
Data streams with multiple values for dataset? I think we should avoid something like that happening at all costs 🤯

Reply:
DLQ? Alert indices?

Reply:
The dlq will also just have a constant_keyword for the dataset such as `_dlq`. To avoid any conflicts that the original document may have with the DLQ metadata (such as @timestamp, error.message, etc.), the original document should be in a dedicated object, such as `event.original`.

I don't know enough about the altering use case.

Reply:
I commented out the fields from this template with a note. These are fields specific to the data stream naming scheme and do not have to be enforced in this template.